pkg/clustermanager/cluster_manager.go

```diff
@@ -277,7 +277,13 @@ func (c *ClusterManager) CreateWorkloadCluster(ctx context.Context, managementCl
 	logger.V(3).Info("Waiting for control plane to be ready")
 	err = c.clusterClient.WaitForControlPlaneReady(ctx, managementCluster, c.controlPlaneWaitTimeout.String(), workloadCluster.Name)
 	if err != nil {
-		return nil, fmt.Errorf("waiting for workload cluster control plane to be ready: %v", err)
+		err = c.Retrier.Retry(
```

Comment:
If this `err` is not nil, we probably should add a warning message and return nil instead of workloadCluster

Reply:
Yeah, i will change the return of workloadCluster to nil. Could the warning message be like "Control plane is not ready, generating workload kubeconfig" ?

Reply:
Well I am referring to if kubeconfig generation fails, so a warning like "attempting to create kubeconfig: <error>"

Reply:
> Well I am referring to if kubeconfig generation fails, so a warning like "attempting to create kubeconfig: "

Okay, I probably need to add that warning after this block( `err=c.Retrier.Retry(     )`), right?